torch/lib/THC/THCTensorMathPointwise.cuh

```diff
@@ -415,11 +415,17 @@ struct TensorDivOp<half> {
 template <typename T>
 struct TensorCRemainderOp {
   __device__ __forceinline__ void operator()(T* out, T* in) {
-    *out = *in != 0 ? *out - *in * (*out / *in) : NAN;
+    *out = (*in != 0) ? *out % *in : NAN;
+    if ((*out * *in)<0){
+      *out += *in;
+    }
   }
 
   __device__ __forceinline__ void operator()(T* out, T* in1, T* in2) {
-    *out = *in2 != 0 ? *in1 - *in2 * (*in1 / *in2) : NAN;
+    *out = (*in2 != 0) ?  *in1 % *in2 : NAN;
```

Comment:
Yeah right now it silently does UB by casting floating point NaN to an integral type, so it's better to raise an error.